lib/models/movie_model.dart

```diff
@@ -0,0 +1,103 @@
+  final String overview;
+  final String releaseDate;
+  final double voteAverage;
+  final List<int> genreIDs;
+  final String pathToBackdropImg;
+  final String pathToPosterImg;
+  final bool adult;
+  final String originalLanguage;
+  final int id;
+  final double popularity;
+  final bool video;
+  final int voteCount;
+
+  const MovieModel({
+    required this.title,
+    required this.originalTitle,
+    required this.overview,
+    required this.releaseDate,
+    required this.voteAverage,
+    required this.genreIDs,
+    required this.pathToBackdropImg,
+    required this.pathToPosterImg,
+    required this.adult,
+    required this.originalLanguage,
+    required this.id,
+    required this.popularity,
+    required this.video,
+    required this.voteCount,
+  });
+
+  factory MovieModel.fromJson(Map<String, dynamic> json) {
+    return MovieModel(
+      title: json['title'],
+      originalTitle: json['original_title'],
+      overview: json['overview'],
+      releaseDate: json['release_date'],
+      voteAverage: json['vote_average'].toDouble(),
+      genreIDs: List<int>.from(
+        (json['genre_ids']),
+      ),
+      pathToBackdropImg: json['backdrop_path'],
+      pathToPosterImg: json['poster_path'],
+      adult: json['adult'],
+      originalLanguage: json['original_language'],
+      id: json['id'],
+      popularity: json['popularity'],
+      video: json['video'],
+      voteCount: json['vote_count'],
+    );
+  }
+
+  factory MovieModel.fromStatic() => const MovieModel(
+        title: 'The Super Mario Bros. Movie',
+        originalTitle: 'The Super Mario Bros. Movie',
+        overview: 'While working underground to fix a water main, Brooklyn '
+            'plumbers—and brothers—Mario and Luigi are transported down a mysterious '
+            'pipe and wander into a magical new world. But when the brothers are separated, '
+            'Mario embarks on an epic quest to find Luigi.',
+        releaseDate: '2023-04-05',
+        voteAverage: 7.8,
+        genreIDs: [
+          12,
+          10751,
+          16,
+          14,
+          35,
+        ],
+        pathToBackdropImg: 'images/SuperMarioBackdrop.jpg',
+        pathToPosterImg: 'images/SuperMarioPoster.jpg',
+        adult: false,
+        originalLanguage: 'en',
+        id: 0,
+        popularity: 100,
+        video: false,
+        voteCount: 100000,
+      );
+
+  Movie getMovieEntity() {
+    List<String> genres = GenreModel.getGenresFromIDs(genreIDs);
+    return Movie(
+        title: title,
+        originalTitle: originalTitle,
+        overview: overview,
+        releaseDate: releaseDate,
+        voteAverage: voteAverage,
+        genres: genres,
+        pathToBackdropImg: pathToBackdropImg,
+        pathToPosterImg: pathToPosterImg,
+        adult: adult,
+        originalLanguage: originalLanguage,
+        id: id,
+        popularity: popularity,
+        video: video,
+        voteCount: voteCount);
+  }
+}
```

Comment:
Do you really need this method?

Reply:
Hmmm, it is true that I call this method after every call to MovieModel's constructors since I only ever work with the entity. I replaced the MovieModel named constructors with static methods that return a Movie entity. Since we never actually use a MovieModel by itself, I also removed every one of its fields. Now it's just a class with no fields that provides static methods which return Movie entities. Is "MovieModel" still an appropriate name?